### PR TITLE
🐛 Source Typeform: add form_id field to records of responses stream

### LINKED ISSUE #21344: Source Typeform: Associate Typeform Response with Form
## Environment
- **Airbyte version**:  0.40.15
- **OS Version / Instance**: AWS EC2
- **Deployment**: Docker
- **Source Connector and version**: Typeform 0.1.9
- **Destination Connector and version**: Snowflake 0.4.38
- **Step where error happened**: Analyzing data

## Current Behavior

- Maybe I'm just missing this, but I can't find any way to associate responses to the forms they were submitted? In looking at [the documentation for the form responses](https://developer.typeform.com/create/reference/retrieve-form) it seems the endpoint itself includes the parameter of the form ID. 

## Expected Behavior
I would expect a table mapping form ID to response ID to count responses for each form. 

## Logs
N/A

## Steps to Reproduce
1. Set up Typeform connector
2. Attempt to filter responses to a specific form

## Are you willing to submit a PR?
Open to helping if there's something small I can do

### PR DESCRIPTION
**Pull request by arsenlosenko**
## What
Resolving issue described here:
https://github.com/airbytehq/airbyte/issues/21344
User requested to have ability to map records from `responses` stream to records from `forms` stream.

## How
Added `form_id` from stream slice that we have and added it to list of records that we receive from `responses` endpoint. Also updated expected records with `form_id` values to match the changed structure of records.

### CODE REVIEW

**Comment by octavia-squidington-iii** [comment]
##  source-typeform test report (commit [`f0df7acc43`](https://github.com/airbytehq/airbyte/pull/27653/commits/f0df7acc43cd2594db93efb4bd42d3f21e20128f)) - ❌

⏲️  Total pipeline duration: 180 seconds

| Step                                                                   | Result   |
|:-----------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-typeform/metadata.yaml | ✅        |
| Connector version semver check.                                        | ✅        |
| Connector version increment check.                                     | ✅        |
| QA checks                                                              | ✅        |
| Code format checks                                                     | ❌        |
| Connector package install                                              | ✅        |
| Build source-typeform docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                             | ✅        |
| Acceptance tests                                                       | ✅        |

🔗 [View the logs here](https://storage.googleapis.com/airbyte-ci-reports/airbyte-ci/connectors/test/pull_request/arsenlosenko_21344-source-typeform-add-form-id-to-response/1687522997/f0df7acc43cd2594db93efb4bd42d3f21e20128f/source-typeform/0.2.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-typeform test
```

**Comment by octavia-squidington-iii** [comment]
##  source-typeform test report (commit [`f0df7acc43`](https://github.com/airbytehq/airbyte/pull/27653/commits/f0df7acc43cd2594db93efb4bd42d3f21e20128f)) - ❌

⏲️  Total pipeline duration: 199 seconds

| Step                                                                   | Result   |
|:-----------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-typeform/metadata.yaml | ✅        |
| Connector version semver check.                                        | ✅        |
| Connector version increment check.                                     | ✅        |
| QA checks                                                              | ✅        |
| Code format checks                                                     | ❌        |
| Connector package install                                              | ✅        |
| Build source-typeform docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                             | ✅        |
| Acceptance tests                                                       | ✅        |

🔗 [View the logs here](https://storage.googleapis.com/airbyte-ci-reports/airbyte-ci/connectors/test/pull_request/arsenlosenko_21344-source-typeform-add-form-id-to-response/1687523271/f0df7acc43cd2594db93efb4bd42d3f21e20128f/source-typeform/0.2.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-typeform test
```

**Comment by octavia-squidington-iii** [comment]
##  source-typeform test report (commit [`5c3925635d`](https://github.com/airbytehq/airbyte/pull/27653/commits/5c3925635d7aafb9cc083086b47cf57958b1511e)) - ✅

⏲️  Total pipeline duration: 157 seconds

| Step                                                                   | Result   |
|:-----------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-typeform/metadata.yaml | ✅        |
| Connector version semver check.                                        | ✅        |
| Connector version increment check.                                     | ✅        |
| QA checks                                                              | ✅        |
| Code format checks                                                     | ✅        |
| Connector package install                                              | ✅        |
| Build source-typeform docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                             | ✅        |
| Acceptance tests                                                       | ✅        |

🔗 [View the logs here](https://storage.googleapis.com/airbyte-ci-reports/airbyte-ci/connectors/test/pull_request/arsenlosenko_21344-source-typeform-add-form-id-to-response/1687523580/5c3925635d7aafb9cc083086b47cf57958b1511e/source-typeform/0.2.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-typeform test
```

File: airbyte-integrations/connectors/source-typeform/Dockerfile [hunk 1]
```diff
@@ -12,5 +12,5 @@ COPY main.py ./
 ENV AIRBYTE_ENTRYPOINT "python /airbyte/integration_code/main.py"
 ENTRYPOINT ["python", "/airbyte/integration_code/main.py"]
 
-LABEL io.airbyte.version=0.2.0
+LABEL io.airbyte.version=0.2.1
```
Comment (davydov-d) [inline airbyte-integrations/connectors/source-typeform/Dockerfile:15]:
```suggestion
LABEL io.airbyte.version=0.3.0
```

**Review by davydov-d — APPROVED** [review]

**Comment by octavia-squidington-iii** [comment]
##  source-typeform test report (commit [`6e500c1786`](https://github.com/airbytehq/airbyte/pull/27653/commits/6e500c1786a050dbf5ddd713cd64a22ec75a08ab)) - ❌

⏲️  Total pipeline duration: 172 seconds

| Step                                                                   | Result   |
|:-----------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-typeform/metadata.yaml | ✅        |
| Connector version semver check.                                        | ✅        |
| Connector version increment check.                                     | ✅        |
| QA checks                                                              | ❌        |
| Code format checks                                                     | ✅        |
| Connector package install                                              | ✅        |
| Build source-typeform docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                             | ✅        |
| Acceptance tests                                                       | ✅        |

🔗 [View the logs here](https://storage.googleapis.com/airbyte-ci-reports/airbyte-ci/connectors/test/pull_request/arsenlosenko_21344-source-typeform-add-form-id-to-response/1687549161/6e500c1786a050dbf5ddd713cd64a22ec75a08ab/source-typeform/0.2.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-typeform test
```

**Comment by octavia-squidington-iii** [comment]
##  source-typeform test report (commit [`87a2f96353`](https://github.com/airbytehq/airbyte/pull/27653/commits/87a2f963538d629fbd0aee361b97a3b4c4992cae)) - ✅

⏲️  Total pipeline duration: 165 seconds

| Step                                                                   | Result   |
|:-----------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-typeform/metadata.yaml | ✅        |
| Connector version semver check.                                        | ✅        |
| Connector version increment check.                                     | ✅        |
| QA checks                                                              | ✅        |
| Code format checks                                                     | ✅        |
| Connector package install                                              | ✅        |
| Build source-typeform docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                             | ✅        |
| Acceptance tests                                                       | ✅        |

🔗 [View the logs here](https://storage.googleapis.com/airbyte-ci-reports/airbyte-ci/connectors/test/pull_request/arsenlosenko_21344-source-typeform-add-form-id-to-response/1687550318/87a2f963538d629fbd0aee361b97a3b4c4992cae/source-typeform/0.3.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-typeform test
```

**Review by lazebnyi — APPROVED** [review]

**Pull request merged**